map.go

```diff
@@ -73,6 +74,17 @@ func (m *BPFMap) ReuseFD(fd int) error {
 	return nil
 }
 
+func (m *BPFMap) AttachStructOps() error {
+	if m.Type().String() != MapTypeStructOps.String() {
+		return errors.New("Map type should be BPF_MAP_TYPE_STRUCT_OPS")
+	}
+	linkC, errno := C.bpf_map__attach_struct_ops(m.bpfMap)
+	if linkC == nil {
+		return fmt.Errorf("Map attach failed: %v", &errno)
```

Comment:
This must follow the pattern used on the lib when the return is handled by libbpf_err_ptr: 

```suggestion
		return fmt.Errorf("failed to attach struct_ops: %w", errno)
```